### PR TITLE
[ENG-1960] Add provider field to registration schemas

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Allows schemas to be associated with a registration provider.

## Changes

- new view RegistrationProviderSchemaList
- new tests for that view
- new migrations

## QA Notes

  - Does this change require a data migration? If so, what data will we migrate?
Small migration
  - What is the level of risk?
Low risk 
    - Any permissions code touched?
No
    - Is this an additive or subtractive change, other?
Additive
  - How can QA verify? (Through UI, API, AdminApp or AdminAdminApp?)
No QA element yet
    - If verifying through API, what's the new version? Please include the endpoints in PR notes or Dev docs.
Endpoint is `v2/providers/registrations/<provider_id>/schemas/`
  - What features or workflows might this change impact?
No workflows effected.
  - How will this impact performance?
No impact
-->

## Documentation

⚠️ Check to update api dev docs when finished.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-1960

### CODE REVIEW

File: api/providers/views.py [hunk 1]
```diff
@@ -595,3 +596,27 @@ def perform_destroy(self, instance):
             self.get_provider().remove_from_group(instance, instance.permission_group)
         except ValueError as e:
             raise ValidationError(str(e))
+
+
+class RegistrationProviderSchemaList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        base_permissions.TokenHasScope,
+        MustBeModerator,
```
Comment (brianjgeiger) [inline api/providers/views.py:605]: This doesn't seem right. Shouldn't everyone have access to the list of schemas available to a registration provider? The use case is that, when you're creating a registration for a registration provider, the front end gets a list of the schemas available for that provider, and then shows that list as possible options to the user.

Comment (Johnetordoff) [reply api/providers/views.py:605]: Ok, this will be totally public, but only for visible and up to date schemas? or all associated schemas?

Comment (brianjgeiger) [reply api/providers/views.py:605]: Hmm. I _think_ visible and up-to-date. Basically whatever you'd expect to see when you're making a registration and it shows you which schemas you have to choose from.

File: api/providers/views.py [hunk 2]
```diff
@@ -595,3 +596,25 @@ def perform_destroy(self, instance):
...
+
+class RegistrationProviderSchemaList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin):
+    permission_classes = (
+        base_permissions.TokenHasScope,
+    )
+    view_category = 'registration-providers'
+    view_name = 'registration-schema-list'
+
+    required_read_scopes = [CoreScopes.MODERATORS_READ]
+    required_write_scopes = [CoreScopes.MODERATORS_WRITE]
```
Comment (brianjgeiger) [inline api/providers/views.py:609]: These scopes are not correct. See `api.schemas.views` for some better options.

File: api/providers/views.py [hunk 3]
```diff
@@ -595,3 +596,25 @@ def perform_destroy(self, instance):
             self.get_provider().remove_from_group(instance, instance.permission_group)
         except ValueError as e:
             raise ValidationError(str(e))
+
+
+class RegistrationProviderSchemaList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin):
+    permission_classes = (
+        base_permissions.TokenHasScope,
```
Comment (brianjgeiger) [inline api/providers/views.py:603]: Probably should also have a `drf_permissions.IsAuthenticatedOrReadOnly,` class here. I'm surprised this isn't causing a test to fail.

Comment (Johnetordoff) [reply api/providers/views.py:603]: yeah it's a issue with testmon where changes in the tests themselves aren't always picked up, it failed for me on my local branch.

File: api_tests/providers/registrations/views/test_registration_provider_schemas.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        return reg_schema
+
+    @pytest.fixture()
+    def provider(self, schema, out_dated_schema, invisible_schema, inactive_schema):
+        provider = RegistrationProviderFactory()
+        update_provider_auth_groups()
+        provider.schemas.add(*[schema, out_dated_schema, invisible_schema, inactive_schema])
+        provider.save()
+        return provider
+
```
Comment (brianjgeiger) [inline api_tests/providers/registrations/views/test_registration_provider_schemas.py:50]: Sorry, one more thing: could you add a test (or just the fixture and a check in the existing test) where a second provider also has just prereg challenge version 2 and make sure that works properly? Amanda is keen on verifying that scenario works, so we might as well have it tested.

File: api/providers/serializers.py [hunk 5]
```diff
@@ -71,7 +71,7 @@ class Meta:
         related_view_kwargs={'provider_id': '<_id>'},
     )
 
-    schemas = TypedRelationshipField(
+    allowed_schemas = TypedRelationshipField(
```
Comment (brianjgeiger) [inline api/providers/serializers.py:74]: Front-end is expecting this relationship to be called, "schemas".

**Pull request merged**